vtex-sites/base.store
# Feat: Style Toast

## What's the purpose of this pull request?

This PR aims to style the Toast component and set its animation (steps for the animation described in [Figma](https://www.figma.com/file/u9z8sTWqBm8eVGsazPLlnX/Handoff-v4-(Regionalization)?node-id=666%3A154997)).

This PR is limited to styled and animate a single Toast. Multiple Toasts will be tackled in [another task](https://vtex-dev.atlassian.net/browse/FSSS-237?atlOrigin=eyJpIjoiMzdmNzAwNDI3ZmQyNDBkMzg2OGUxZjdhOWExMTk0NTkiLCJwIjoiaiJ9).

## How does it work?

Every time a new Toast is generate by using the [pushToast](https://github.com/vtex-sites/base.store/blob/3c31e5b72c4937cc53a7f4e873ff6aad63d23a12/src/sdk/ui/index.tsx#L38), the `popToast` will trigger the Toast component and it will be displayed. 

Also, `Toasts` are being generated whenever a new notification arrives from [useCart](https://github.com/vtex-sites/base.store/blob/3c31e5b72c4937cc53a7f4e873ff6aad63d23a12/src/components/common/Toast/Toast.tsx#L20). This behavior needs to be revised, since messages lack some properties such as `icon` and `description`. Maybe this is the case to use `Alerts` instead of `Toasts`. 

## How to test it?

Just for quick test purposes, I've created this temporary `add toast` button to push a `Toast` ai see it in action. This should be removed and a proper example must be created in the Storybook. Just heads up on that, since `Toast` has some SDK logic in it, Storybook complained when I tried to create an example. 

![toast2](https://user-images.githubusercontent.com/10616243/165013270-154a9827-2d71-4efb-8fa1-462046b79df6.gif)

## References

Related [Jira task](https://vtex-dev.atlassian.net/browse/FSSS-231)
Related [Figma file](https://www.figma.com/file/u9z8sTWqBm8eVGsazPLlnX/Handoff-v4-(Regionalization)?node-id=666%3A154997)

## Checklist

- [ ] Style Toast
- [ ] Animate Toast
- [ ] Add Toast to Storybook
